matrix-sdk-android/src/main/java/org/matrix/android/sdk/api/failure/Extensions.kt

```diff
@@ -53,6 +53,16 @@ fun Throwable.toRegistrationFlowResponse(): RegistrationFlowResponse? {
                     .adapter(RegistrationFlowResponse::class.java)
                     .fromJson(this.errorBody)
         }
+    } else if (this is Failure.ServerError && this.error.code == MatrixError.M_FORBIDDEN) {
```

Comment:
You do not test the httpCode here? Maybe not mandatory and not specced though...

Reply:
Yes it should be 401, i can add the check
> If the homeserver decides that an attempt on a stage was unsuccessful, but the client may make a second attempt, it returns the same HTTP status 401 response as above, with the addition of the standard errcode and error fields describing the error.